src/Network/Email/Email.php

```diff
@@ -1885,4 +1886,44 @@ protected function _getContentTypeCharset()
         }
         return strtoupper($this->charset);
     }
+
+    /**
+     * Serializes the email object to a value that can be natively serialized and re-used
+     * to clone this email instance.
+     *
+     * @return array Serializable array of configuration properties.
+     */
+    public function jsonSerialize()
+    {
+        $properties = [
+            '_to', '_from', '_replyTo', '_cc', '_bcc', '_subject', '_returnPath', '_readReceipt',
+            '_template', '_layout', '_viewRender', '_viewVars', '_theme', '_helpers', '_emailFormat', '_transport',
+            '_attachments', '_domain', '_messageId', '_headers', 'charset', 'headerCharset',
+        ];
+
+        $array = [];
+
+        foreach ($properties as $property) {
+            $array[$property] = $this->{$property};
+        }
+
+        return array_filter($array);
+    }
+
+    /**
+     * Configures an email instance object from serialized config.
+     *
+     * @param string $json JSON encoded email configuration.
+     * @return \Cake\Network\Email\Email Configured email instance.
+     */
+    public function jsonConfig($json)
```

Comment:
One last thing about this @lorenzo - it's a JSON string that is passed to it and then decoded, should i let one pass an array instead? Not sure what's considered to be the 'right way'.